aea/cli/eject.py

```diff
@@ -135,10 +133,49 @@ def _eject_item(ctx: Context, item_type: str, public_id: PublicId):
             break
     supported_items.add(new_public_id)
     supported_items.remove(present_public_id)
-    update_item_config("agent", Path(ctx.cwd), **{item_type_plural: supported_items})
+    component_type = ComponentType(item_type)
+    _update_custom_configurations(
+        ctx.agent_config.component_configurations,
+        ComponentId(component_type, present_public_id),
+        ComponentId(component_type, new_public_id),
+    )
+    update_item_config(
```

Comment:
these could be better named, especially the second one, as it's not clear which item config? First one should not be private.